src/nnue/nnue_feature_transformer.h

```diff
@@ -289,11 +294,16 @@ namespace Eval::NNUE {
     // Calculate cumulative value using difference calculation
     void UpdateAccumulator(const Position& pos) const {
 
-      const auto prev_accumulator = pos.state()->previous->accumulator;
+      Accumulator* prev_accumulator;
+      if (pos.state()->previous->accumulator.computed_accumulation)
+        prev_accumulator = &pos.state()->previous->accumulator;
+      else
+        prev_accumulator = &pos.state()->previous->previous->accumulator;
```

Comment:
is this `pos.state()->previous->previous` always associated (also near the root?). Maybe good to put an assert there?

Reply:
Yes, to reach here without `pos.state()->previous->accumulator.computed_accumulation` being true, `pos.state()->previous->previous` must exist. I'll add assert.